### PR TITLE
Update zarr creation task

### PR DESCRIPTION
**Pull request by mckinsel**
1. Sort labels so we get the same order in every matrix
2. Split numeric and string cell metadata
3. Match the spec
    a. Set shapes and chunks according to spec
    b. Flatten out "expression_matrix" group

Please ensure the following when opening a PR:
- [ ] Added or updated tests
- [ ] Updated documentation
- [ ] Applied style guidelines
- [ ] Considered generalizability beyond our own use case

### CODE REVIEW

**Comment by dshiga** [comment]
retest this please

File: docker/python3-scientific/create_zarr_ss2.py [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+            pipeline output, separated by commas
+        rsem_genes_results_file (str): the file for the expression count
+        output_zarr_path (str): location of the output zarr
+        file_format (str): zarr file format  [DirectoryStore or ZipStore ] Default: DirectoryStore
+    """
+    # initiate the zarr file
+    root_group = init_zarr(sample_id, output_zarr_path, file_format)
+
+    # add the the gene/TMP counts
+    read_and_convert_expression(root_group, rsem_genes_results_file)
```
[the remark this answers is not in the document]
Comment (mckinsel) [reply docker/python3-scientific/create_zarr_ss2.py]: Yeah, one of the goals here was to remove the intermediate `expression_matrix` group that contained everything.

File: docker/python3-scientific/create_zarr_ss2.py [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+        compressor=COMPRESSOR,
+        dtype="f4",
+        chunks=(1, len(numeric_metadata)),
+        data=[sorted_numeric_values]
+    )
+
+    # Cell IDs
+    data_group.create_dataset(
+        "cell_id",
+        shape=(1,),
```
[the remark this answers is not in the document]
Comment (mckinsel) [reply docker/python3-scientific/create_zarr_ss2.py]: Just matching the spec that was expecting this to be one dimensional. Note that `len(QC_values[2][0:1])` must always evaluate to 0 or 1 (or `IndexError`...), so this is mostly replacing `(1, 1)` with `(1,)`.

File: docker/python3-scientific/create_zarr_ss2.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+    if file_format == "ZipStore":
+        store = zarr.ZipStore(path, mode='w')
+
+    # create the root group
+    root = zarr.group(store, overwrite=True)
+
+    # add some readme for the user
+    root.attrs['README'] = ("The schema adopted in this zarr store may undergo "
+                            "changes in the future")
+    root.attrs['sample_id'] = sample_id
```
[the remark this answers is not in the document]
Comment (mckinsel) [reply docker/python3-scientific/create_zarr_ss2.py]: I'm not sure. Would that just make us match the rest of the metadata better?

File: docker/python3-scientific/create_zarr_ss2.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (rexwangcc) [reply docker/python3-scientific/create_zarr_ss2.py]: Yeah that makes sense to me then! I was thinking about the shape in numpy, where `(1, 1)`  is not the same as `(1,)` 😄

[the remark this answers is not in the document]
Comment (mckinsel) [reply docker/python3-scientific/create_zarr_ss2.py]: Yeah it's not the same in zarr either (which mostly matches all numpy semantics), so this is indeed a change.

File: docker/python3-scientific/create_zarr_ss2.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (rexwangcc) [reply docker/python3-scientific/create_zarr_ss2.py]: So in the metadata-api library(which is designed to be the python api of metadata in the future), it's using sequencing_input for the sample_id: https://github.com/HumanCellAtlas/metadata-api/blob/c889bc86752254cdfa8825145bf8a1020753603b/src/humancellatlas/data/metadata/api.py#L628

But refactor this name will require us to change a lot of places in the pipeline, I think it's better to leave this change separate from this PR!

**Pull request merged**